concent_api/core/tests/test_payments_bankster.py

```diff
@@ -92,176 +140,103 @@ class FinalizePaymentBanksterTest(ConcentIntegrationTestCase):
-                        subtask_cost=self.subtask_cost,
-                    )
-
-        self.assertIsInstance(requestors_claim_payment_info, ClaimPaymentInfo)
-        self.assertIsInstance(providers_claim_payment_info, ClaimPaymentInfo)
+    def test_that_when_there_are_other_deposit_claims_finalize_payment_substract_them_from_currently_processed_claim(self):
+        self.deposit_claim = DepositClaim()
+        self.deposit_claim.subtask_id = self._get_uuid('1')
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+        self.deposit_claim.payee_ethereum_address = self.task_to_compute.provider_ethereum_address
+        self.deposit_claim.concent_use_case = ConcentUseCase.FORCED_ACCEPTANCE
+        self.deposit_claim.amount = 2
+        self.deposit_claim.clean()
+        # Save twice because we want two same claims.
+        self.deposit_claim.save()
+        self.deposit_claim.pk = None
+        self.deposit_claim.subtask_id = self._get_uuid('2')
+        self.deposit_claim.save()
 
-        self.assertIsNotNone(requestors_claim_payment_info.tx_hash)
-        self.assertIsNotNone(requestors_claim_payment_info.payment_ts)
-        self.assertEqual(requestors_claim_payment_info.amount_paid, self.subtask_cost)
-        self.assertEqual(requestors_claim_payment_info.amount_pending, 0)
+        with mock.patch('core.payments.service.get_deposit_value', return_value=5) as get_deposit_value:
+            with mock.patch('core.payments.service.force_subtask_payment', return_value=MOCK_TRANSACTION) as force_subtask_payment:
+                returned_value = finalize_payment(self.deposit_claim)
 
-        self.assertIsNone(providers_claim_payment_info.tx_hash)
-        self.assertIsNone(providers_claim_payment_info.payment_ts)
-        self.assertEqual(providers_claim_payment_info.amount_paid, 0)
-        self.assertEqual(providers_claim_payment_info.amount_pending, 0)
+        self.assertEqual(returned_value, encode_hex(MOCK_TRANSACTION.hash))
 
-        self.assertEqual(get_deposit_value.call_count, 1)
-        self.assertEqual(
-            get_deposit_value.call_args_list[0][1],
-            {'client_eth_address': self.task_to_compute.requestor_ethereum_address}
+        get_deposit_value.assert_called_with(
+            client_eth_address=self.deposit_claim.payer_deposit_account.ethereum_address,
         )
         force_subtask_payment.assert_called_once_with(
-            requestor_eth_address=self.task_to_compute.requestor_ethereum_address,
-            provider_eth_address=self.task_to_compute.provider_ethereum_address,
-            value=self.subtask_cost,
-            subtask_id=self.task_to_compute.subtask_id,
+            requestor_eth_address=self.deposit_claim.payer_deposit_account.ethereum_address,
+            provider_eth_address=self.deposit_claim.payee_ethereum_address,
+            # 5 - (2 + 2) | deposit_value - sum_of_other_claims
+            value=1,
```

Comment:
is `5` `TaskToCompute`'s cost? If so, I would suggest setting it explicitly in setup

Reply:
No, it is in mock in line 224.